<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TransactDialog</class>
 <widget class="QDialog" name="TransactDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>604</width>
    <height>695</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Transact</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="10" column="2">
    <widget class="QPushButton" name="debug">
     <property name="text">
      <string>&amp;Debug</string>
     </property>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="QSpinBox" name="gas">
     <property name="specialValueText">
      <string>Automatic</string>
     </property>
     <property name="suffix">
      <string> gas</string>
     </property>
     <property name="minimum">
      <number>-1</number>
     </property>
     <property name="maximum">
      <number>430000000</number>
     </property>
     <property name="value">
      <number>-1</number>
     </property>
    </widget>
   </item>
   <item row="3" column="3">
    <widget class="QComboBox" name="valueUnits"/>
   </item>
   <item row="4" column="3">
    <widget class="QComboBox" name="gasPriceUnits"/>
   </item>
   <item row="2" column="1" colspan="3">
    <widget class="QLineEdit" name="calculatedName">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
     <property name="placeholderText">
      <string/>
     </property>
    </widget>
   </item>
   <item row="7" column="1">
    <widget class="QCheckBox" name="optimize">
     <property name="text">
      <string>&amp;Optimise</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="10" column="3">
    <widget class="QPushButton" name="send">
     <property name="text">
      <string>&amp;Execute</string>
     </property>
     <property name="default">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="7" column="3">
    <widget class="QSpinBox" name="nonce">
     <property name="enabled">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="8" column="0" colspan="4">
    <widget class="QSplitter" name="splitter_5">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="QPlainTextEdit" name="data">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="lineWidth">
       <number>0</number>
      </property>
     </widget>
     <widget class="QTextEdit" name="code">
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="lineWidth">
       <number>0</number>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
   </item>
   <item row="3" column="1" colspan="2">
    <widget class="QSpinBox" name="value">
     <property name="suffix">
      <string/>
     </property>
     <property name="maximum">
      <number>430000000</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
   </item>
   <item row="10" column="1">
    <widget class="QPushButton" name="copyUnsigned">
     <property name="text">
      <string>Copy &amp;Unsigned</string>
     </property>
    </widget>
   </item>
   <item row="0" column="1" colspan="3">
    <widget class="QComboBox" name="from"/>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="label_6">
     <property name="text">
      <string>&amp;Gas</string>
     </property>
     <property name="buddy">
      <cstring>gas</cstring>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label5">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>&amp;To</string>
     </property>
     <property name="buddy">
      <cstring>destination</cstring>
     </property>
    </widget>
   </item>
   <item row="10" column="0">
    <widget class="QPushButton" name="cancel">
     <property name="text">
      <string>&amp;Cancel</string>
     </property>
     <property name="shortcut">
      <string>Esc</string>
     </property>
    </widget>
   </item>
   <item row="7" column="2">
    <widget class="QCheckBox" name="autoNonce">
     <property name="text">
      <string>Auto Nonce</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <property name="autoRepeat">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="1" column="1" colspan="3">
    <widget class="QComboBox" name="destination">
     <property name="editable">
      <bool>true</bool>
     </property>
     <item>
      <property name="text">
       <string>(Create Contract)</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="9" column="0" colspan="4">
    <widget class="QLabel" name="total">
     <property name="sizePolicy">
      <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label5_2">
     <property name="text">
      <string>&amp;Amount</string>
     </property>
     <property name="buddy">
      <cstring>value</cstring>
     </property>
    </widget>
   </item>
   <item row="7" column="0">
    <widget class="QLabel" name="label_2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>D&amp;ata</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
     <property name="buddy">
      <cstring>data</cstring>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>&amp;From</string>
     </property>
     <property name="buddy">
      <cstring>from</cstring>
     </property>
    </widget>
   </item>
   <item row="4" column="2">
    <widget class="QSpinBox" name="gasPrice">
     <property name="prefix">
      <string>@ </string>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>430000000</number>
     </property>
    </widget>
   </item>
   <item row="5" column="3">
    <widget class="QSpinBox" name="maxGas">
     <property name="suffix">
      <string> gas</string>
     </property>
     <property name="prefix">
      <string>max </string>
     </property>
     <property name="maximum">
      <number>450000000</number>
     </property>
    </widget>
   </item>
   <item row="5" column="2">
    <widget class="QSpinBox" name="minGas">
     <property name="readOnly">
      <bool>true</bool>
     </property>
     <property name="suffix">
      <string> gas</string>
     </property>
     <property name="prefix">
      <string>min </string>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>450000000</number>
     </property>
    </widget>
   </item>
   <item row="5" column="1">
    <widget class="QProgressBar" name="progressGas">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="value">
      <number>24</number>
     </property>
     <property name="format">
      <string>Calculating gas...</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>autoNonce</sender>
   <signal>toggled(bool)</signal>
   <receiver>nonce</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>374</x>
     <y>196</y>
    </hint>
    <hint type="destinationlabel">
     <x>451</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
